File: mindy/week1/J-9375.py
# 조합 사용
# 각 조합을 1개만 하는 경우도 고려해야하기 때문에 +1
# 마지막으로 알몸인 경우는 제외하므로 -1
# dictionary의 요소로 list 사용가능하므로 append

import sys

input = sys.stdin.readline


n = int(input().strip())
for i in range(n):
    dic=dict()
    m = int(input().strip())
    for _ in range(m):
        a, b = input().strip().split()
        if b in dic:
            dic[b].append(a)
        else:
            dic[b]=[a]
    

    cnt=1
    for k in dic:
        cnt *= (len(dic[k])+1)

    print(cnt-1)

File: leo/week1/J-9375.js
/**
 *  제목 : 패션왕 신해빈
 *  입력 : T(1 ≤ T ≤ 100)
 *  제한사항 : X
 *  시간계산 필요 : O(T * n), 최대 T=100, n=30의 경우를 고려한 시간복잡도 계산
 */

// Run by Node.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const input = [];

rl.on("line", (line) => {
  input.push(line);
}).on("close", () => {
  const T = parseInt(input[0]); // 테스트 케이스의 수
  let result = [];

  let index = 1;
  for (let i = 0; i < T; i++) {
    const n = parseInt(input[index]);
    let clothesMap = {};

    for (let j = 1; j <= n; j++) {
      const [name, type] = input[index + j].split(" ");

      if (clothesMap[type]) {
        clothesMap[type]++;
      } else {
        clothesMap[type] = 1;
      }
    }

    let combinations = 1;

    for (let key in clothesMap) {
      combinations *= clothesMap[key] + 1;
    }

    // 알몸이 아닌 경우만 계산
    result.push(combinations - 1);

    index += n + 1;
  }

  console.log(result.join("\n"));
  process.exit();
});
